app/models/concerns/with_organization_status/base.rb

```diff
@@ -0,0 +1,22 @@
+class WithOrganizationStatus::Base
```

Comment:
Here we are mixing two different hierarchies: the `WithOrganizationStatus`  concern, which is for things - organizations? - that have a status, and the concept of organization's lifecycle  itself. We should split those concepts, since the latter is by no means a concern. I think this hierarchy should be reamed `OrganizationStatus`

I really like the idea of reifying the organization's status. We should ensure it is properly introduced into the codebase.